mongodb_store/src/mongodb_store/util.py

```diff
@@ -391,6 +415,27 @@ def update_message(collection, query_doc, msg, meta, upsert):
 
     # convert msg to db document
     doc=msg_to_document(msg)
+
+    if hasattr(msg, 'pose'):
```

Comment:
These are repeated field checks for object insertion and object update. But of course I can create a method instead of repeating the check.